src/components/TopBar/tests/TopBar.test.tsx

```diff
@@ -200,38 +204,36 @@ describe('<TopBar />', () => {
     );
 
     it('renders', () => {
-      const topBar = mountWithContext(
+      const topBar = mountWithAppProvider(
         <TopBar contextControl={mockContextControl} />,
       );
       expect(findByTestID(topBar, 'ContextControl').exists()).toBe(true);
       expect(topBar.contains(mockContextControl)).toBe(true);
     });
 
     it('doesn’t render a logo when defined', () => {
-      const topBar = mountWithContext(
+      const topBar = mountWithAppProvider(
         <TopBar contextControl={mockContextControl} />,
-        {
+        mergeThemeProviderContext({
           logo: {
             topBarSource: './assets/shopify.svg',
           },
-        },
+        }),
       );
       expect(topBar.find(Image).exists()).toBe(false);
     });
 
     it('doesn’t render the wrapper when not defined and no logo is available', () => {
-      const topBar = mountWithContext(<TopBar />);
+      const topBar = mountWithAppProvider(<TopBar />);
       expect(findByTestID(topBar, 'ContextControl').exists()).toBe(false);
     });
   });
 });
 
-function mountWithContext(
-  node: React.ReactElement<any>,
-  polarisTheme?: ThemeContext,
-) {
-  const context = polarisTheme ? {polarisTheme} : createThemeContext();
-  return mountWithAppProvider(
-    <ThemeProvider value={context}>{node}</ThemeProvider>,
-  );
+function mergeThemeProviderContext(providedThemeContext: ThemeProviderContext) {
+  return {
+    context: {
+      themeProvider: merge({logo: null}, providedThemeContext),
+    },
+  };
```

Comment:
Tell me more about `mergeThemeProviderContext`

Reply:
It's just a level of abstraction.  Rather than repeating ⬇ every time
```tsx
{
  context: {
    themeProvider: {
      logo: ...
    }
  }
}
```

 we can use `mergeThemeProviderContext({logo: ...})`, and if we ever change the object's shape in the future we'll only have to change `mergeThemeProviderContext`